### DISCUSSION TITLE
[BUG]: uploadListingImage 2.0 does not all ALT tags

### DISCUSSION THREAD

**Discussion by moneticc** [etsy/open-api#713] [Q&A]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

We are trying to mass update the Alt text for all of our existing images with API, but we experience error if there are all 10 pictures used in the listing. In short - we use POST:

```
POST
"/application/shops/20714867/listings/1112913521/images"
array:3 [
  "listing_image_id" => 3485780781
  "alt_text" => "Blue ALT tag"
  "rank" => 1
]
```
And here is the response:

**Received HTTP status code [400] with error "Listings are only allowed 10 images. Please delete a ListingImage before trying to upload another.".**

The problem here is that you will see this error if the listing has all 10 listing images previously uploaded. There is no error returned if there are 9 or fewer images in the listing. In the case of 9 or fewer images, it is possible to update Alt texts for all 9 images, but when there are 10 images it returns an error that I wrote previously.

The thing I want to point out is that we do not upload new images, we are just updating Alt text for already existing images.

Please advise how we can update the Alt text for all ten images (we simply do not want to delete 1 image and then upload it once again)?

Etsy's initial response was: **_If there are already 10 images on the listing and you want to update the alt text, you'll need to pass in the image_id, the alt text and set overwrite as true._**

In such case the code would be as follow:

```
METHOD: POST
URL: https://api.etsy.com/v3/application/shops/20714867/listings/1112913521/images
PARAMS: 
[ 
"listing_image_id" => 4090513898 
"alt_text" => "Blue ALT tag" 
"rank" => 1 
"overwrite" => true 
]
```

But the thing is that if we do like this it does not add an alt tag, but simply deletes the picture. So, basically, it is not possible to add ALT tag if there is all 10 pictures used. Anyone can help out?

### Steps to reproduce

Anyone can easily reproduce by adding 10 pictures to any listing and trying out to update the alt tags.

### Expected behavior

Without **"overwrite" => true** simply brings back an error, but with **"overwrite" => true**  deletes the picture.

**Comment by etsyachristensen** [comment]
@moneticc We are still tracking this issue. We are closing the issue board in favor of the Discussions board so we'll convert this issue to a new discussion. Will continue tracking there.

**Comment by cyrilchandelier** [comment]
@moneticc did you ever find a workaround for this? 

I'm experiencing the exact same issue:
- Error 400 "Listings are only allowed 10 images. Please delete a ListingImage before trying to upload another." when I have 10 images but only try to update the alt text
- The image is completely deleted if I set `overwrite: true`

**Comment by sushansht** [comment]
Any updates in this issue?
I'm facing the same problem.

when updating an image in a certain rank with overwrite true, only an image is updated but the alt text would not be updated. Since, new image means new alt text it would be wonderful if alt text is overwritten too.

Only workaround is to delete existing images and reupload the new images.

Please fix this issue 🙏

**Comment by moneticc** [comment]
No updates guys, seems that the issue still exists.
